keps/sig-cli/3805-ssa-default/README.md

```diff
@@ -0,0 +1,567 @@
+
+In the context of this change, a rollback would possibly be someone
+switching from server-side apply to client-side apply and vice-versa.
+This problem isn't new and is one of the reason that motivates this
+change. It mostly works well when the default fieldmanager is being
+used.
+
+<<[UNRESOLVED bad description of the problems here ]>>
+We could certainly make a much better analysis of the problems we have
+migrating from server-side to client-side and vice-versa here.
+<<[/UNRESOLVED]>>
+
+###### What specific metrics should inform a rollback?
+
+People should inform their decision based on the direct error they get
+from kubectl as they are trying to apply their resources.
+
+###### Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?
+
+We have a series of tests for that.
+
+<<[UNRESOLVED find the tests ]>>
+I need to add the tests here.
+<<[/UNRESOLVED]>>
+
+###### Is the rollout accompanied by any deprecations and/or removals of features, APIs, fields of API types, flags, etc.?
+
+This will eventually change the default behavior of `kubectl apply` in
+possibly surprising way. `kubectl apply` can now fail because of a
+conflict, which it wouldn't before on newly created objects. Another
+surprising behavior is that since fields can be owned by multiple
+actors, removing a field will not necessarily remove the field from the
+cluster's resource.
+
+### Monitoring Requirements
+
+N/A. No monitoring in place.
+
+###### How can an operator determine if the feature is in use by workloads?
+
+`kubectl apply -V8` can help identify what type of apply is used. Also
+checking the presence of the last applied annotation on the cluster
+resource indicates whether the resource was client-side or server-side
+applied.
+
+###### How can someone using this feature know that it is working for their instance?
+
+- [x] Other (treat as last resort)
+  - Details: Kubectl will fail with an error right away.
+
+###### What are the reasonable SLOs (Service Level Objectives) for the enhancement?
+
+N/A.
+
+###### What are the SLIs (Service Level Indicators) an operator can use to determine the health of the service?
+
+N/A.
+
+###### Are there any missing metrics that would be useful to have to improve observability of this feature?
+
+N/A.
+
+### Dependencies
+
+###### Does this feature depend on any specific services running in the cluster?
+
+Feature depends on server-side apply being enabled in the cluster, but
+the feature has been GA since 1.22 and is now guaranteed to be present
+in all clusters supported by the kubectl version (1 version schew).
+
+### Scalability
+
+###### Will enabling / using this feature result in any new API calls?
+
+For now, we will continue to get resources before server-side applying,
+which means that the number of requests will be similar (one GET, one
+PATCH). This is done because we need to verify if the resource needs to
+be client-side applied or server-side applied.
```

Comment:
If we can fully remove CSA, we can get rid of that initial GET request, right?

Reply:
absolutely